data_safe_haven/pulumi/components/sre_backup.py

```diff
@@ -0,0 +1,193 @@
+"""Pulumi component for SRE state"""
+import datetime
+
+from pulumi import ComponentResource, Input, ResourceOptions
+from pulumi_azure_native import dataprotection, resources
+
+
+class SREBackupProps:
+    """Properties for SREBackupComponent"""
+
+    def __init__(
+        self,
+        location: Input[str],
+        storage_account_securedata_id: Input[str],
+        storage_account_securedata_name: Input[str],
+    ) -> None:
+        self.location = location
+        self.storage_account_securedata_id = storage_account_securedata_id
+        self.storage_account_securedata_name = storage_account_securedata_name
+
+
+class SREBackupComponent(ComponentResource):
+    """Deploy SRE backup with Pulumi"""
+
+    def __init__(
+        self,
+        name: str,
+        stack_name: str,
+        props: SREBackupProps,
+        opts: ResourceOptions | None = None,
+    ) -> None:
+        super().__init__("dsh:sre:BackupComponent", name, {}, opts)
+        child_opts = ResourceOptions.merge(opts, ResourceOptions(parent=self))
+        start_date = str(datetime.datetime.now(datetime.timezone.utc).date())
```

Comment:
`start_date` will change every time the code is run. Won't that cause Pulumi to make unnecessary changes and make it appear the state is incorrect?

What happens if the time ISO repeating time period just starts are some date in the past?